rasa_nlu/training_data/loading.py

```diff
@@ -57,26 +53,35 @@ def load_data(resource_name, language='en'):
         return data_sets[0].merge(*data_sets[1:])
 
 
+def _reader_factory(fformat):
+    """Generates the appropriate reader class based on the file format."""
+    reader = None
+    if fformat == LUIS:
+        reader = LuisReader()
+    elif fformat == WIT:
+        reader = WitReader()
+    elif fformat in DIALOGFLOW_RELEVANT:
+        reader = DialogflowReader()
+    elif fformat == RASA:
+        reader = RasaReader()
+    elif fformat == MARKDOWN:
+        reader = MarkdownReader()
+    return reader
+
+
 def _load(filename, language='en'):
     """Loads a single training data file from disk."""
 
     fformat = _guess_format(filename)
+    if fformat == UNK:
+        raise ValueError("Unknown data format for file {}".format(filename))
 
     logger.info("Training data format of {} is {}".format(filename, fformat))
-
-    if fformat == LUIS:
-        return LuisReader().read(filename)
-    elif fformat == WIT:
-        return WitReader().read(filename)
-    elif fformat.startswith("dialogflow"):
-        return _from_dialogflow_file(filename, language, fformat)
-    elif fformat == RASA:
-        return RasaReader().read(filename)
-    elif fformat == MARKDOWN:
-        return MarkdownReader().read(filename)
-    else:
-        raise ValueError("unknown training file format : {} for "
-                         "file {}".format(fformat, filename))
+    reader = _reader_factory(fformat)
+    td = None
+    if reader:
+        td = reader.read(filename, language=language, fformat=fformat)
+    return td
```

Comment:
MIN: I'd rather go for (don't like the poorly named temp var)
```
if reader:
    return reader.read(filename, language=language, fformat=fformat)
else:
    return None
```
But this might be personal preference